resources/views/components/editable.blade.php

```diff
@@ -0,0 +1,39 @@
+{{-- components.editable
+
+Render an editable input field
+
+--}}
+
+<div
+        x-data="{
+            field: '{{ $field }}',
+            id: {{ $model->id }},
+            value: {{ json_encode($model->$field) }},
+            original: {{ json_encode($model->$field) }},
+            editing: false
+        }"
+        @click.away="editing = false; value = original;"
+>
+        <input
+                x-cloak
+                x-ref="input"
+                x-show="editing"
+                x-model="value"
+                @keydown.enter="
+                    $wire.update({
+                    id: id,
+                    value: value,
```

Comment:
I was thinking that what if we send something like `$wire.update(id, { [field]: value })` so we can use it directly on the Livewire component like

```php
function update($id, $data)
{
    // $data would be something like ['name' => 'My new name']
    Model::find($id)->update($data);
}
```

since `$data` would have the associative array the model needs, we also could have access to the model instance using a type hint like

```php
function update(User $user, $data)
{
    $user->update($data);
}
```

what do you think?

Reply:
I'm reaching the point where I think it will be better if you finally do it.
I mean, I have no problem doing it, but I think it will be faster if you use my draft and apply it by yourself in the way you want.
It's faster to do it than actually to write the comments here.
I'm open, I can do it when I can, or you can just finish the idea and implement yourself.